### PR TITLE
Adding compareMemWithTwoFiles to FileReadUtility for Future SW

### PR DESCRIPTION
**Pull request by mlarson02**
Added a function to FileReadUtility.h, compareMemWithTwoFiles, which will be needed as a dependency for an upcoming PR in order to properly check the IRProducer / IRFakeProducer memories in the IRAnalyer. The existing function, compareMemWithFile, cannot be used in checkIRMemories, as two reference files are needed for each instance of computed memories. This function could presumably be used in any other cases in which two memory reference files are needed to be compared with an instance of computed memories.

### CODE REVIEW

File: TestBenches/FileReadUtility.h [hunk 1]
```diff
@@ -413,6 +413,79 @@ class TBHelper {
...
+  unsigned int err_count = 0;
+
+  ////////////////////////////////////////
+  // Read from file
+  MemType memory_ref;
+  MemType memory_ref1;
+  MemType memory_ref2;
+  writeMemFromFile<MemType>(memory_ref1, *foutVec[0], ievt, InputBase);
+  if (foutVec.size() > 1){
+  writeMemFromFile<MemType>(memory_ref2, *foutVec[1], ievt, InputBase);
```
Comment (aehart) [inline TestBenches/FileReadUtility.h:431]: @mlarson02 Please indent this line two spaces.

File: TestBenches/FileReadUtility.h [hunk 2]
```diff
@@ -413,6 +413,79 @@ class TBHelper {
...
+  }
+  for (int i = 0; i<memory_ref1.getEntries(0); ++i){
+    auto data_ref1 = memory_ref1.read_mem(0, i);
+    memory_ref.write_mem(0, data_ref1, i);
+  }
+  if (foutVec.size() > 1){
+  for (int j = 0; j < memory_ref2.getEntries(0); ++j){
+    auto data_ref2 = memory_ref2.read_mem(0, j);
+    memory_ref.write_mem(0, data_ref2, j + memory_ref1.getEntries(0));
+  }
```
Comment (aehart) [inline TestBenches/FileReadUtility.h:441]: @mlarson02 Please indent these lines two spaces.

File: TestBenches/FileReadUtility.h [hunk 3]
```diff
@@ -413,6 +413,79 @@ class TBHelper {
...
+    } else if (data_com != data_ref) {
+      std::cout << "\t" << "<=== INCONSISTENT";
+      err_count++;
+    }
+
+    std::cout << std::endl;
+  }
+  
+  return err_count;
+
```
[the remark this answers is not in the document]
Comment (mlarson02) [reply TestBenches/FileReadUtility.h]: Just addressed these issues in the latest push and verified that compareMemWithFile and compareMemWithTwoFiles still work on the future SW side. Let me know if these changes look good or if there's anything else that needs to change. Thanks!

[the remark this answers is not in the document]
Comment (mlarson02) [reply TestBenches/FileReadUtility.h]: Fixed these issues, and verified both functions still work with future SW, in my most recent push. Let me know if there is anything else that needs to be changed/done, thanks!

**Comment by aehart** [comment]
@mlarson02 Could you try rebasing on the master branch? GitHub seems confused by the order of the commits and is showing a bunch of changes that you haven't made.

Assuming you are in your local clone of firmware-hls and on the filereadutility_futuresw branch, the commands are:
```bash
git fetch --all
git rebase origin/master
git diff origin/filereadutility_futuresw # just to verify that nothing actually changed
git push --force origin filereadutility_futuresw
```

**Comment by mlarson02** [comment]
> @mlarson02 Could you try rebasing on the master branch? GitHub seems confused by the order of the commits and is showing a bunch of changes that you haven't made.
> 
> Assuming you are in your local clone of firmware-hls and on the filereadutility_futuresw branch, the commands are:
> 
> ```shell
> git fetch --all
> git rebase origin/master
> git diff origin/filereadutility_futuresw # just to verify that nothing actually changed
> git push --force origin filereadutility_futuresw
> ```

@aehart I Just ran all of these commands. When running ` git rebase origin/master` and  `git push --force origin filereadutility_futuresw` it says that everything is up to date, as you can see in the terminal output I attached. 
image

**Review by aehart — APPROVED** [review]
> > @mlarson02 Could you try rebasing on the master branch? GitHub seems confused by the order of the commits and is showing a bunch of changes that you haven't made.
> > Assuming you are in your local clone of firmware-hls and on the filereadutility_futuresw branch, the commands are:
> > ```shell
> > git fetch --all
> > git rebase origin/master
> > git diff origin/filereadutility_futuresw # just to verify that nothing actually changed
> > git push --force origin filereadutility_futuresw
> > ```
> 
> @aehart I Just ran all of these commands. When running ` git rebase origin/master` and `git push --force origin filereadutility_futuresw` it says that everything is up to date, as you can see in the terminal output I attached. image

That's strange. I ran the same commands, but the rebase for me actually did something. Maybe it's down to different git versions…

Anyway, I took the liberty of pushing the results of my rebase. No files were actually changed, but the commit history is more sensible now. And I'm happy with your changes, so I'll approve the PR.

**Pull request merged**